www/js/services/unifiedDataLoader.ts

```diff
@@ -0,0 +1,68 @@
+import { getRawEntries } from './commHelper';
+import { ServerResponse, ServerData, TimeQuery } from '../types/serverData';
+
+/**
+ * removeDup is a helper function for combinedPromises
+ * @param list An array of values from a BEMUserCache promise
+ * @returns an array with duplicate values removed
+ */
+export const removeDup = function (list: Array<ServerData<any>>) {
+  return list.filter(function (value, i, array) {
+    const firstIndexOfValue = array.findIndex(function (element) {
+      return element.metadata.write_ts == value.metadata.write_ts;
+    });
+    return firstIndexOfValue == i;
+  });
+};
+
+export const combinedPromises = function (
+  promiseList: Array<Promise<any>>,
+  filter: (list: Array<any>) => Array<any>,
+) {
+  if (promiseList.length === 0) {
+    throw new RangeError('combinedPromises needs input array.length >= 1');
+  }
+  return new Promise(function (resolve, reject) {
+    Promise.allSettled(promiseList).then(
+      (results) => {
+        var allRej = true;
+        var values = [];
+        var rejections = [];
```

Comment:
```suggestion
        let allRej = true;
        const values = [];
        const rejections = [];
```

Reply:
I had no idea a const array could still be changed with its member functions, interesting!

Reply:
Yeah `const` just means it can't be reassigned. So you can modify `const` objects all you want as long as that object keeps the same address in memory.